### PR TITLE
custom cd pipeline

### PR DESCRIPTION
**Pull request by averyschreiner**

### CODE REVIEW

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - name: Build project
+      run: dotnet publish -c Release -o app
+
+    - name: Deploy to EC2
+      env:
+        PRIVATE_KEY: ${{ secrets.EC2_PRIVATE_KEY }}
+        HOST_NAME: ${{ secrets.EC2_HOST_NAME }}
+        USER_NAME: ${{ secrets.EC2_USER_NAME }}
+      run: |
+        echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
```
Comment (Pchristy03) [inline .github/workflows/pipeline.yml]: should it be chmod 400?

File: .github/workflows/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        ssh -i private_key -o StrictHostKeyChecking=no ${USER_NAME}@${HOST_NAME} `
```
Comment (Pchristy03) [inline .github/workflows/pipeline.yml]: What is the `StrictHostKeyChecking`?

File: .github/workflows/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    - name: Install dependencies
+      run: dotnet restore
+
+    - name: Build solution
+      run: dotnet build
+
+    - name: Run tests
+      run: |
+        dotnet test Chefster.Tests/Chefster.Tests.csproj
```
Comment (Pchristy03) [inline .github/workflows/pipeline.yml:30]: Will it stop the deploy if unit tests fail? I assume it probably will

Comment (averyschreiner) [reply .github/workflows/pipeline.yml:30]: yeah it won't run the 'deploy_to_ec2' job if the 'run_unit_test' job fails.
peep line 33

Comment (Pchristy03) [reply .github/workflows/pipeline.yml:30]: ope my bad, nice 👍🏻

File: .github/workflows/pipeline.yml (hunk 2 above)
Comment (averyschreiner) [reply .github/workflows/pipeline.yml]: image

Comment (averyschreiner) [reply .github/workflows/pipeline.yml]: we might remove that part actually

File: .github/workflows/pipeline.yml (hunk 1 above)
Comment (averyschreiner) [reply .github/workflows/pipeline.yml]: we sure can!

File: .github/workflows/pipeline.yml (hunk 2 above)
Comment (Pchristy03) [reply .github/workflows/pipeline.yml]: well we actually might need that since the instance IP address can change. What is the HOST_NAME configured as?

**Pull request merged**